cmd/registry_login_test.go

```diff
@@ -0,0 +1,150 @@
+package cmd
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func Test_GenerateRegistryAuth(t *testing.T) {
+	registryURL := "https://index.docker.io/v1/"
+	username := "docker_user"
+	password := "docker_password"
+	gotBytes, _ := generateRegistryAuth(registryURL, username, password)
+	got, err := bytesToRegistryStruct(gotBytes)
+
+	if err != nil {
+		t.Errorf("Error converting bytes to struct, %q", err)
+		t.Fail()
+	}
+
+	want := RegistryAuth{
+		AuthConfigs: map[string]Auth{
+			registryURL: {Base64AuthString: base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", username, password)))},
+		},
+	}
+
+	if !reflect.DeepEqual(got, want) {
```

Comment:
How would someone fix this error without seeing the value of either struct printed?